src/Sylius/Component/Product/spec/Model/ProductSpec.php

```diff
@@ -120,6 +122,66 @@ function it_refuses_to_remove_non_product_attribute(AttributeValueInterface $att
         $this->shouldThrow('\InvalidArgumentException')->duringRemoveAttribute($attribute);
     }
 
+    function it_returns_attributes_by_locale(
+        ProductAttributeInterface $attribute,
+        ProductAttributeValueInterface $attributeValueEN,
+        ProductAttributeValueInterface $attributeValuePL
+    ): void
+    {
+        $attribute->getCode()->willReturn('colour');
+
+        $attributeValueEN->setProduct($this)->shouldBeCalled();
+        $attributeValueEN->getLocaleCode()->willReturn('en_US');
+        $attributeValueEN->getAttribute()->willReturn($attribute);
+        $attributeValueEN->getCode()->willReturn('colour');
+        $attributeValueEN->getValue()->willReturn('Blue');
+
+        $attributeValuePL->setProduct($this)->shouldBeCalled();
+        $attributeValuePL->getLocaleCode()->willReturn('pl_PL');
+        $attributeValuePL->getAttribute()->willReturn($attribute);
+        $attributeValuePL->getCode()->willReturn('colour');
+        $attributeValuePL->getValue()->willReturn('Niebieski');
+
+        $this->addAttribute($attributeValueEN);
+        $this->addAttribute($attributeValuePL);
+
+        $this
+            ->getAttributesByLocale('pl_PL', 'en_US')
+            ->toArray()
+            ->shouldReturn([$attributeValuePL->getWrappedObject()])
+        ;
+    }
+
+    function it_returns_attributes_by_locale_and_base_locale(
+        ProductAttributeInterface $attribute,
+        ProductAttributeValueInterface $attributeValueEN,
+        ProductAttributeValueInterface $attributeValueFR
+    ): void
+    {
+        $attribute->getCode()->willReturn('colour');
+
+        $attributeValueEN->setProduct($this)->shouldBeCalled();
+        $attributeValueEN->getLocaleCode()->willReturn('en_US');
+        $attributeValueEN->getAttribute()->willReturn($attribute);
+        $attributeValueEN->getCode()->willReturn('colour');
+        $attributeValueEN->getValue()->willReturn('Blue');
+
+        $attributeValueFR->setProduct($this)->shouldBeCalled();
+        $attributeValueFR->getLocaleCode()->willReturn('fr_FR');
+        $attributeValueFR->getAttribute()->willReturn($attribute);
+        $attributeValueFR->getCode()->willReturn('colour');
+        $attributeValueFR->getValue()->willReturn('Bleu');
+
+        $this->addAttribute($attributeValueEN);
+        $this->addAttribute($attributeValueFR);
+
+        $this
+            ->getAttributesByLocale('pl_PL', 'en_US', 'fr_FR')
+            ->toArray()
+            ->shouldReturn([$attributeValueEN->getWrappedObject()])
```

Comment:
Let's add another ones which are testing the same call but:
 - returning the attribute from `$localeCode` locale
 - returning the attribute from `$baseLocaleCode` locale